### DISCUSSION TITLE
Adding the Roles-based enablement of Setting

### DISCUSSION THREAD

**Discussion by mafrasiabi** [open-webui/open-webui#597] [General]
hi there,

thank you so much for this fascinating repo.

I was wondering whether it's possible to add a feature where only admin could have access to some of the settings e.g. adding a model, exporting and importing the prompts, etc. and the extent of access for different users can be managed?

**Comment by justinh-rahb** [comment]
Currently only admins can manage models, modelfiles and prompts, which is in line with your request. Expanding further from that, I'm sure that having more granular permissions and groups would be useful for the future.

**Comment by mafrasiabi** [comment]
thanks for the quick answer, is there a place that admins can change the prompt for RAG part only?

**Comment by justinh-rahb** [comment]
> thanks for the quick answer, is there a place that admins can change the prompt for RAG part only?

@mafrasiabi at the moment, no, there doesn't appear to be a specific prompt in the backend RAG code but maybe there's a default in the libraries it uses. I think this would be a worthy reason to open another issue separately as that would be a useful feature on its own, or at least Timothy can explain how it works and close it 👍

**Comment by justinh-rahb** [comment]
Related: #446

**Comment by tjbck** [comment]
@mafrasiabi  I'm not entirely sure what you mean by "changing the prompt for RAG part" but you could modify the rag prompt template here: [src/lib/utils/rag/index.ts](https://github.com/ollama-webui/ollama-webui/blob/main/src/lib/utils/rag/index.ts)

**Comment by flefevre** [comment]
Hello, thanks for your amazing work at open webui.

I do agree that the two role admin and user are not enough.
I think it will be preferable to have admain, user and manager.
A manager can have access to the prompt library, the document library and the model customization but not the new model calling.

I have look at anythingllm product, they have introduce a call feature which is "workspace" a shared sapce with document between identified users.

**Comment by flefevre** [comment]
Finally how will it be implemented in the next release?
Did you chose to introduce a 'manager' ' role additionally to admin and user?
How did you design it?
